### PR TITLE
Reader: add post notifications toggle to settings popover in development

### PR DESCRIPTION
**Pull request by bluefuton**
Adds a feature flag `reader/new-post-notifications` and, where enabled, uses the new `ReaderSiteNotificationSettings` block rather than `ReaderEmailSettings`.

The UI is not final and the toggle for new posts notifications is not yet functional. It'll be connected up in subsequent PRs.

See https://github.com/Automattic/wp-calypso/issues/19440.

screen shot 2017-12-14 at 13 11 11

### Why a new block?

I started adding conditionals to `ReaderEmailSettings` based on the feature flag, but decided that the layout and functionality of the new popover is sufficiently different to merit a new block.

It'd need a rename at some point anyway, because the new block contains both email and push notification settings.

### To test

In development mode, ensure that you are shown the new post notification toggle in the popover from the Reader feed header and from Manage Following.

In other environments, ensure that you are shown the existing email settings popover.

### CODE REVIEW

File: client/blocks/reader-subscription-list-item/index.jsx [hunk 1]
```diff
@@ -43,7 +45,7 @@ function ReaderSubscriptionListItem( {
 	className = '',
 	translate,
 	followSource,
-	showEmailSettings,
+	showEmailSettings, // @todo rename to showNotificationSettings
```
Comment (bluefuton) [inline client/blocks/reader-subscription-list-item/index.jsx]: I'll address this in a followup. I've made a note of it :)

File: client/blocks/reader-feed-header/index.jsx [hunk 2]
```diff
@@ -50,12 +52,19 @@ class FeedHeader extends Component {
...
 		const siteUrl = getSiteUrl( { feed, site } );
 		const isEmailBlocked = userSettings.getSetting( 'subscription_delivery_email_blocked' );
+		const siteId = site && site.ID;
 
 		const classes = classnames( 'reader-feed-header', {
 			'is-placeholder': ! site && ! feed,
 			'has-back-button': showBack,
 		} );
 
+		const notificationSettings = config.isEnabled( 'reader/new-post-notifications' ) ? (
```
Comment (blowery) [inline client/blocks/reader-feed-header/index.jsx]: before we would never create these elements if `siteId` was falsey. Moving it out here changes that a bit... Can these elements handle a null siteId?

File: client/blocks/reader-site-notification-settings/index.jsx [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+			? this.props.unsubscribeToNewCommentEmail
+			: this.props.subscribeToNewCommentEmail;
+
+		toggleSubscription( this.props.siteId );
+	};
+
+	render() {
+		const { translate, notifyOnNewComments, notifyOnNewPosts } = this.props;
+
+		if ( ! this.props.siteId ) {
```
Comment (blowery) [inline client/blocks/reader-site-notification-settings/index.jsx]: yup, looks like they can handle a null / 0 siteId. :)

File: client/blocks/reader-site-notification-settings/index.jsx [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+						<FormToggle onChange={ this.toggleNewCommentEmail } checked={ notifyOnNewComments } />
+					</div>
+				</ReaderPopover>
+			</div>
+		);
+	}
+}
+
+const mapStateToProps = ( state, ownProps ) => {
+	const follow = find( getReaderFollows( state ), { blog_ID: ownProps.siteId } );
```
Comment (blowery) [inline client/blocks/reader-site-notification-settings/index.jsx]: hmmm... if siteId is falsey, should we do the rest of the work?

File: config/test.json [hunk 5]
```diff
@@ -90,6 +90,7 @@
 		"reader/following-intro": true,
 		"reader/full-errors": true,
 		"reader/list-management": false,
+		"reader/new-post-notifications": false,
```
Comment (blowery) [inline config/test.json]: might want it true here if we're going to write tests around it

Comment (bluefuton) [reply config/test.json]: Yes! Good point.

File: client/blocks/reader-site-notification-settings/index.jsx (hunk 4 above)
Comment (bluefuton) [reply client/blocks/reader-site-notification-settings/index.jsx]: 👍  I'll return early if siteId is falsey.

**Pull request merged**